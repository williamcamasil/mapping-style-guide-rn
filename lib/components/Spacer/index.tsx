import React, { useMemo } from 'react';
import { Text, ViewStyle } from 'react-native';

type SpacerPropsType = {
  size: ViewStyle['height'];
};

const Spacer: React.FC<SpacerPropsType> = ({ size }) => {
  const textStyle = useMemo(() => ({
    height: size,
    width: size,
  }), [size]);
  /*
  * A decisão de usar um Text em vez de um View é para que esse Spacer
  * possa ser usado dentro de outras Views e tbm de outros Text sem restrição.
  */
  return (
    <Text style={textStyle} />
  );
};

export default Spacer;
