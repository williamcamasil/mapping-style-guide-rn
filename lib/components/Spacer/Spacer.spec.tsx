import React from 'react';

import renderer from 'react-test-renderer';

import Spacer from '.';

describe('Spacer snapshot', () => {
  it('default', () => {
    const tree = renderer
      .create(
        <Spacer size={28} />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
