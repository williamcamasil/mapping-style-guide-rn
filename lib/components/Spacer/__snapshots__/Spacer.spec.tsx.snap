// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spacer snapshot default 1`] = `
<Text
  style={
    {
      "height": 28,
      "width": 28,
    }
  }
/>
`;
